src/test/java/io/jenkins/plugins/sprp/AgentTest.java

```diff
@@ -0,0 +1,160 @@
+
+        agentSnippetActual =
+                pipelineSnippetGenerator.autoAddTabs((ArrayList<String>) pipelineSnippetGenerator.getAgent(agent));
+
+        agentSnippetExpected =
+                "agent {\n" +
+                        "\tdocker {\n" +
+                        "\t\timage 'my-docker-image'\n" +
+                        "\t\targs '-v temp:temp'\n" +
+                        "\t\talwaysPull true\n" +
+                        "\t\treuseNode true\n" +
+                        "\t}\n" +
+                        "}\n";
+
+        assertEquals(agentSnippetExpected, agentSnippetActual);
+
+        agent.setAlwaysPull(false);
+        agent.setReuseNode(false);
+
+        agentSnippetActual =
+                pipelineSnippetGenerator.autoAddTabs((ArrayList<String>) pipelineSnippetGenerator.getAgent(agent));
+
+        agentSnippetExpected =
+                "agent {\n" +
+                        "\tdocker {\n" +
+                        "\t\timage 'my-docker-image'\n" +
+                        "\t\targs '-v temp:temp'\n" +
+                        "\t\talwaysPull false\n" +
+                        "\t\treuseNode false\n" +
+                        "\t}\n" +
+                        "}\n";
+
+        assertEquals(agentSnippetExpected, agentSnippetActual);
+    }
+
+
+    @Test
+    public void dockerfileGenerationTest() {
+        Agent agent = new Agent();
+        agent.setDockerfile("my-dockerfile");
+        agent.setArgs("-v temp:temp");
+
+        String agentSnippetActual =
+                pipelineSnippetGenerator.autoAddTabs((ArrayList<String>) pipelineSnippetGenerator.getAgent(agent));
+
+        String agentSnippetExpected =
+                "agent {\n" +
+                        "\tdockerfile {\n" +
+                        "\t\tfilename 'my-dockerfile'\n" +
+                        "\t\tadditionalBuildArgs '-v temp:temp'\n" +
+                        "\t\treuseNode false\n" +
+                        "\t}\n" +
+                        "}\n";
+
+        assertEquals(agentSnippetExpected, agentSnippetActual);
+
+        agent.setLabel("my-label");
+        agent.setCustomWorkspace("my-customWorkspace");
+        agent.setDir("dir-path");
+        agent.setReuseNode(true);
+
+        agentSnippetActual =
+                pipelineSnippetGenerator.autoAddTabs((ArrayList<String>) pipelineSnippetGenerator.getAgent(agent));
+
+        agentSnippetExpected =
+                "agent {\n" +
+                        "\tdockerfile {\n" +
+                        "\t\tfilename 'my-dockerfile'\n" +
+                        "\t\tdir 'dir-path'\n" +
+                        "\t\tadditionalBuildArgs '-v temp:temp'\n" +
+                        "\t\tlabel 'my-label'\n" +
+                        "\t\tcustomWorkspace 'my-customWorkspace'\n" +
+                        "\t\treuseNode true\n" +
+                        "\t}\n" +
+                        "}\n";
+
+        assertEquals(agentSnippetExpected, agentSnippetActual);
```

Comment:
it's generally not good to have multiple assertions in the same test. If your goal is to specifically test that changing the agent and re-generating creates a different snippet, than that is ok.

but my guess from looking at this is that it should be split into three different test cases. I make that assumption because you're changing the `agentSnippetExpected` variable, which indicates that the test's expectation changes.